specs/ledger/hs/src/Ledger/Update.hs

```diff
@@ -539,3 +542,244 @@ instance STS UPEND where
 
 instance Embed FADS UPEND where
   wrapFailed = error "No possible failures in FADS"
+
+------------------------------------------------------------------------
+-- Update interface
+------------------------------------------------------------------------
+
+-- | The update interface environment is shared amongst various rules, so
+--   we define it as an alias here.
+type UPIEnv =
+  ( Core.Epoch
+  , Core.Slot
+  , Map Core.VKeyGenesis Core.VKey
+  )
+
+-- | The update interface state is shared amongst various rules, so we define it
+-- as an alias here.
+type UPIState =
```

Comment:
I never thought you'd buy into this idea of using tuples for the states :)

Reply:
It fits really well with the latex spec!